src/common/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -97,21 +97,31 @@ static bool eliminate_reshape_v1(const std::shared_ptr<Node>& node) {
     if (input.get_shape() == node->get_output_shape(0)) {
         return replace_output_update_name(node->output(0), input);
     }
+
     // eliminate redundant reshape, squeeze, or unsqueeze
     auto input_node = input.get_node_shared_ptr();
+    if (input_node->get_output_size() != 1)
```

Comment:
It doesn't check the number of consumers, as `get_output_size` returns number of output ports but not consumers. Please use `get_target_inputs` after you casted input node to one of this reshape like operations.

Reply:
And please add a tests for this.